src/hooks/useIsGoalImpossible.tsx

```diff
@@ -0,0 +1,34 @@
+import { getChildrenGoals } from "@src/api/GoalsAPI";
+import { getImpossibleGoalById } from "@src/api/ImpossibleGoalsApi";
+import { useEffect, useState } from "react";
+
+const useIsGoalImpossible = ({ id }: { id: string }) => {
```

Comment:
this is a little too costly. we can't create a hook that runs for ever goal every time a single change in DOM will trigger this. can we think of some better approach ?

Reply:
Hi @Tushar-4781, I am unable to find an alternative solution. Could you share any ideas or suggestions?

Reply:
i think this involves some refactoring in components. can you resolve failing tests of #1837 cuz i did some refactoring already. post that we can think of a better structure cuz it little difficult for me to sync in.

Reply:
@Tushar-4781, currently we are passing goals as props through multiple components. As a first step toward refactoring can we store the fetched goals in the recoil state and then directly use that state in the required component?